src/lib/AST/BitcodeWriter.cpp

```diff
@@ -1116,6 +1126,32 @@ emitBlock(
     emitBlock(I.FriendType);
 }
 
+void
+BitcodeWriter::
+emitBlock(
+    AliasInfo const& I)
+{
+    StreamSubBlockGuard Block(Stream, BI_ALIAS_BLOCK_ID);
+    emitInfoPart(I);
+    emitSourceInfo(I);
+    if (I.AliasedSymbol)
+        emitBlock(*I.AliasedSymbol);
+}
+
+void
+BitcodeWriter::
+emitBlock(
+    UsingInfo const& I)
+{
+    StreamSubBlockGuard Block(Stream, BI_USING_BLOCK_ID);
+    emitInfoPart(I);
+    emitSourceInfo(I);
+    emitRecord(I.UsingSymbols, USING_SYMBOLS);
+    if (I.Qualifier)
+        emitBlock(*I.Qualifier);
+    emitRecord(I.Class, USING_CLASS);
```

Comment:
I would emit `USING_CLASS` after `emitSourceInfo`.. just in case we decide to store different things for different kinds of using declarations